### PR TITLE
fix the case mapping_accepts=dict and add_sources=smth

### LINKED ISSUE #387: Mapping won't work with mapping_accepts=dict and add_sources
@dmitriy-serdyuk , there was another option in `Mapping` transofrmer, and I think it's incompatible now with the new feature that you introduced.

### PR DESCRIPTION
**Pull request by rizar**
fix #387

### CODE REVIEW

File: fuel/transformers/__init__.py [hunk 1]
```diff
@@ -225,14 +225,17 @@ def get_data(self, request=None):
...
             raise ValueError
         data = next(self.child_epoch_iterator)
+        mapping_input = data
         if self.mapping_accepts == dict:
-            data = OrderedDict(equizip(self.data_stream.sources, data))
-        image = self.mapping(data)
+            mapping_input = OrderedDict(equizip(self.data_stream.sources,
+                                                data))
+        image = self.mapping(mapping_input)
+        image_sources = self.add_sources if self.add_sources else self.sources
```
Comment (dmitriy-serdyuk) [inline fuel/transformers/__init__.py]: It is easier to read when the default case goes first.

File: fuel/transformers/__init__.py [hunk 2]
```diff
@@ -225,14 +225,17 @@ def get_data(self, request=None):
...
         if self.mapping_accepts == dict:
-            image = tuple(image[source] for source in self.sources)
-        if not self.add_sources:
-            return image
-        return data + image
+            image = tuple(image[source] for source in image_sources)
+        if self.add_sources:
+            return data + image
+        return image
```
Comment (dmitriy-serdyuk) [inline fuel/transformers/__init__.py]: Same here.

File: fuel/transformers/__init__.py (hunk 1 above)
Comment (rizar) [inline fuel/transformers/__init__.py]: I remember reading that having conditions like `if not smth` is a bad style.

Comment (dmitriy-serdyuk) [inline fuel/transformers/__init__.py]: You probably mean `if not x is None`. My common sense tells me that the `else` body is exceptional and rare.

But it is not a big concern.

**Comment by nouiz** [comment]
I think it was for Python 2.4 support.

Le mer. 19 avr. 2017 15:44, Dzmitry Bahdanau <notifications@github.com> a
écrit :

**Comment by dmitriy-serdyuk** [comment]
@rizar , you want to fix the test? And I will merge it.

**Pull request merged**